packages/integration-tests/test/actions.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { Config } from '../../../common'
+import { Web3Provider } from '@ethersproject/providers'
+import { ganache } from '@eth-optimism/ovm-toolchain'
+import assert = require('assert')
+
+const {
+  Contract,
+  ContractFactory,
+  providers: { JsonRpcProvider },
+  Wallet,
+} = require('ethers');
+const fs = require('fs')
+
+const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))
+
+let SimpleStorage
+let L2Messenger
+const L1_USER_PRIVATE_KEY = process.env.L1_USER_PRIVATE_KEY
+const L2_USER_PRIVATE_KEY = process.env.L2_USER_PRIVATE_KEY
+const goerliURL = process.env.L1_URL
+const optimismURL = process.env.L2_URL
+const l1Provider = new JsonRpcProvider(goerliURL)
+const l2Provider = new JsonRpcProvider(optimismURL)
+const l1Wallet = new Wallet(L1_USER_PRIVATE_KEY, l1Provider)
+const l2Wallet = new Wallet(L2_USER_PRIVATE_KEY, l2Provider)
+const messengerJSON = JSON.parse(fs.readFileSync('contracts/build/iOVM_BaseCrossDomainMessenger.json'))
+const l2MessengerJSON = JSON.parse(fs.readFileSync('contracts/build/OVM_L2CrossDomainMessenger.json'))
+
+const deploySimpleStorage = async () => {
+  const SimpleStorageJson = JSON.parse(fs.readFileSync('contracts/build/SimpleStorage.json'))
+  const SimpleStorageFactory = new ContractFactory(SimpleStorageJson.abi, SimpleStorageJson.bytecode, l2Wallet)
+  SimpleStorage = await SimpleStorageFactory.deploy()
+  console.log('Deployed SimpleStorage to', SimpleStorage.address)
+}
+
+const deposit = async (amount) => {
```

Comment:
This function should accept more arguments so that the assertions don't appear to assert on magic numbers

Reply:
I'm not totally sure what you mean by this. What other arguments are you thinking? Or are you referring to making the function variadic?

Reply:
The second argument here:

```js
const calldata = SimpleStorage.interface.encodeFunctionData('setValue', [`0x${'42'.repeat(32)}`])
```
is hardcoded in the `deposit` function and in the assertion. To understand that the test is correct, a reviewer must read through and notice that the deposit function hardcodes the same value that is in the assertion below. It would make it much easier for reviewers if the `deposit` function accepted an argument `value` that was used in place of the hardcoded `0x...` so that the same value can be passed to the assert statement

Reply:
Ah totally that makes sense